packages/account-v2/src/components/LeaveConfirm/LeaveConfirm.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useFormikContext } from 'formik';
+import { useHistory } from 'react-router-dom';
+import { Button, Modal, Text, useDevice } from '@deriv-com/ui';
+import UnsavedChanges from '../../assets/status-message/ic-unsaved-changes.svg';
+
+type TLeaveConfirm = {
+    onCancel?: () => void;
+    onLeave?: () => void;
+};
+
+export const LeaveConfirm = ({ onCancel, onLeave }: TLeaveConfirm) => {
+    const history = useHistory();
+    const { dirty } = useFormikContext();
+    const { isMobile } = useDevice();
+
+    const [showPrompt, setShowPrompt] = useState(false);
+    const [currentPath, setCurrentPath] = useState('');
+    const unblock = useRef<ReturnType<ReturnType<typeof useHistory>['block']>>();
+
+    useEffect(() => {
+        unblock.current = history.block((prompt: { pathname: React.SetStateAction<string> }) => {
+            if (dirty) {
+                setCurrentPath(prompt.pathname);
+                setShowPrompt(true);
+            }
+            return !dirty;
+        });
+        return () => {
+            unblock.current();
+        };
+    }, [dirty, history]);
+
+    const handleLeave = () => {
+        onLeave?.();
+        unblock.current();
+        setShowPrompt(false);
+        history.push(currentPath);
+    };
+
+    const handleCancel = () => {
+        onCancel?.();
+        setShowPrompt(false);
+    };
+
+    return (
+        <Modal className='w-[440px]' isOpen={showPrompt}>
```

Comment:
border radius and h-auto is already applied on the Modal component, padding and width seems to be the only things that needs to be applied. We can look into apply but i think we should let the Modal have there own width property since it might vary in other places.
![image](https://github.com/binary-com/deriv-app/assets/135801848/8d91ee28-5dd0-4db7-ad5c-92dd9333e953)

Reply:
Cant we ask a variant in deriv-com/ui?
i think all the modals have curved edges.

Reply:
For desktop curved edges are already applied in the Modal component in deriv-com ui. I think for width we should stick with separate style for individual Modal implementation because it might vary.

Reply:
The modal styles must have updated in some version. This wasn't there earlier. Lets remove the duplicated styles?

Reply:
Updated.